com.unity.render-pipelines.high-definition/Runtime/Sky/SkyManager.cs

```diff
@@ -744,7 +816,10 @@ internal void RequestStaticEnvironmentUpdate()
                 var renderingContext = cachedContext.renderingContext;
 
                 if (IsCachedContextValid(skyContext))
+                {
                     forceUpdate |= skyContext.skyRenderer.DoUpdate(m_BuiltinParameters);
+                    forceUpdate |= (skyContext.HasClouds() && skyContext.cloudRenderer.DoUpdate(m_BuiltinParameters));
```

Comment:
Just to make sure I get this right, this will force an update only if parameters of the cloud layer used in the "GeBakingHashCode" function change right? But not all the parameters are in this function right? 
What about the other parameters? It seems to me that the GetHashCode of the cloud settings should also be incorporated in the hash that we compute for the sky and use the existing control flow. This way if any of the parameters other than the "baking" ones that could change the look of the cloud will also trigger a recompute if the sky is set on "OnChanged"

Reply:
i included the GetHashCode of the clouds inside the hash of the sky (above in ComputeSkyHash), that way the code doesn't need much change and the update mode is controlled like before from the sky dropdown

Reply:
Oh yeah missed it because I was looking at ComputeCloudHash that does not seem to be used at all. Might want to remove it if it's not necessary.
Also I'm still concerned that GetHashCode is not overridden at all in CloudLayer.cs so an "OnChanged" sky will probably not update cubemap correctly when changing the cloud layer parameters.

Reply:
Nice catch for ComputeCloudHash.
The update is working when using OnChanged, because the volume component default GetHashCode function computes the hash of all parameters combined. It's easier when iterating but now I can implement it properly

Reply:
Ah yeah forgot about that, thanks :)
It could be better to make it explicit indeed because sometimes we don't want ALL parameters to actually participate in the hash .